From 8a6c940a9a1dc9517f5db5ca97c165948e49e023 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Sat, 28 Oct 2023 00:21:23 +0530
Subject: [PATCH] md: dm-crypt: Fix DUN size to 4 for inline crypt

SDHCI CQ supports only 32 DUN bits and max_dun_size should be 4.

Fix iv_size to 4 for inline crypt which is passed as dun_bytes for
blk_crypto_init_key.

Change-Id: I022011d64f5e2985241aafcea5f07429cfb6c518
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/md/dm-crypt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/dm-crypt.c b/drivers/md/dm-crypt.c
index f5102a2c94d3..19672b5d157e 100644
--- a/drivers/md/dm-crypt.c
+++ b/drivers/md/dm-crypt.c
@@ -2394,7 +2394,7 @@ static int crypt_select_inline_crypt_mode(struct dm_target *ti, char *cipher,
 	}
 
 	if (ivmode == NULL || (strcmp(ivmode, "plain64") == 0)) {
-		cc->iv_size = 8;
+		cc->iv_size = 4;
 	} else {
 		ti->error = "Invalid IV mode for inline_crypt";
 		return -EINVAL;
-- 
2.34.1

